src/main/java/org/apache/maven/it/Embedded3xLauncher.java

```diff
@@ -268,4 +271,26 @@ public String getMavenVersion()
         throw new LauncherException( "Could not determine embedded Maven version" );
     }
 
+    /**
+     * Replicates the logic from <a href="https://git.io/JSMug">Maven start script</a> to find 
+     * the project's base directory.
+     * @param workingDirectory the working directory
+     * @return the project's base directory
+     */
+    private static File findProjectBaseDirectory( File workingDirectory )
```

Comment:
Attention, this could be confused with `${project.basedir}` which is not the same. In the open JIRA ticket for a new property I have proposed something like `project.executionRoot.basedir`. Would it make sense to use this term here as well?

Reply:
Well, this is not necessarily the execution root, because the directory usually stays the same no matter whether you really execute it on the project root or just for a sub module (in case a .mvn directory is found). Also this is how the init script calls it, therefore I would prefer keeping this name

Reply:
maybe `findProjectRootDirectory`?

Reply:
Yes, you are right. If this is an aggregator it will stop. Leave it as-is for now, thanks for the correctin.